app/src/main/java/com/epfl/drawyourpath/utils/Utils.kt

```diff
@@ -392,4 +394,118 @@ object Utils {
     fun coordinatesToBitmap(coordinates: List<LatLng>, size: Int = 100, paint: Paint = defaultPaint): Bitmap {
         return strokesToBitmap(listOf(coordinatesToStroke(coordinates)), size, paint)
     }
+
+    /**
+     * Reduces the number of points in a Path.
+     * @param path The path we want to reduce
+     * @param maxError The max error percentage relative to the path distance
+     * @return A similar Path with the least useful points removed.
+     */
+    fun reducePath(path: Path, maxError: Float = 0.01F): Path {
+        val reducedPoints = mutableListOf(listOf<LatLng>())
+        if (path.size() <= 2) {
+            return Path(path.getPoints())
+        }
+
+        val epsilon = maxError * path.getDistance().toFloat()
+        for (section in path.getPoints()) {
+            reducedPoints.add(reduceSection(section, epsilon))
+        }
+        reducedPoints.removeFirst()
+        return Path(reducedPoints.toList())
+    }
```

Comment:
I think this function can be simplified a bit by returning `path` when the algorithm is not used, and by infering the type of `reducedPoint` instead of adding an empty list.
```suggestion
    fun reducePath(path: Path, maxError: Float = 0.01F): Path {
        val reducedPoints = mutableListOf<List<LatLng>>()
        if (path.size() <= 2) {
            return path
        }

        val epsilon = maxError * path.getDistance().toFloat()
        for (section in path.getPoints()) {
            reducedPoints.add(reduceSection(section, epsilon))
        }
        return Path(reducedPoints.toList())
    }
```

Reply:
Right I forgot that we do not need to instanciate the list directly.
For the path I wanted to make a copy but that is not necessary.

Your suggestion works perfectly then